### DISCUSSION TITLE
What are the use cases for inspectFields?

### DISCUSSION THREAD

**Discussion by villesau** [urql-graphql/urql#2712] [Q&A]
Hey, I've read about inspectFields from here: https://formidable.com/open-source/urql/docs/api/graphcache/#inspectfields and https://formidable.com/open-source/urql/docs/graphcache/cache-updates/#updating-many-unknown-links

Are there any other realistic use cases than inspecting the `Query` fields and filtering over to the desired field? I could not find any other usage examples for that. If there are, would be cool if there could be some examples documented in https://formidable.com/open-source/urql/docs/api/graphcache/#inspectfields

**Comment by JoviDeCroock** [comment]
Pagination is a big one, it's used internally to implement [our pagination resolvers](https://github.com/FormidableLabs/urql/blob/222395206cb6da32a9391c9a0a06004fc3466b79/exchanges/graphcache/src/extras/relayPagination.ts#L193).

Another example might be if you have a user which has todo's, you create a todo you might want to invalidate all todo-lists for that user which would go something like

```js
const user = { __typename: 'User', id }
const fields = cache.inspectFields(user)
for (const field of fields) {
  if (field.fieldName === 'todoList')
    cache.invalidate(user, field.fieldKey)
}
```

So basically things that are unknown in the context of your resolver can now be achieved, i.e. all previously fetched lists can be invalidated
